packages/opentelemetry-semantic-conventions/src/resource/SemanticResourceAttributes.ts

```diff
@@ -14,155 +14,837 @@
+import {
+  DOT,
+  UNDERSCORE,
+  ACCOUNT_DOT,
+  CLOUD_DOT,
+  CONTAINER_DOT,
+  CLUSTER_DOT,
+  DEPLOYMENT_DOT,
+  DEVICE_DOT,
+  ECS_DOT,
+  EKS_DOT,
+  EXECUTABLE_DOT,
+  FAAS_DOT,
+  GROUP_DOT,
+  K8S_DOT,
+  LOG_DOT,
+  IMAGE_DOT,
+  MODEL_DOT,
+  NODE_DOT,
+  OS_DOT,
+  PROCESS_DOT,
+  RUNTIME_DOT,
+  SDK_DOT,
+  SERVICE_DOT,
+  SET_DOT,
+  STREAM_DOT,
+  TASK_DOT,
+  TELEMETRY_DOT,
+  WEBENGINE_DOT,
+  ARN,
+  AWS,
+  AZURE,
+  COMMAND,
+  CRON,
+  DAEMON,
+  DESCRIPTION,
+  HOST,
+  ID,
+  INSTANCE,
+  JOB,
+  NAME,
+  REPLICA,
+  SPACE,
+  STATEFUL,
+  TYPE,
+  UID,
+  VERSION,
+} from '../internal/constants';
+
+//----------------------------------------------------------------------------------------------------------
 // DO NOT EDIT, this is an Auto-generated file from scripts/semconv/templates//templates/SemanticAttributes.ts.j2
-export const SemanticResourceAttributes = {
+//----------------------------------------------------------------------------------------------------------
+
+//----------------------------------------------------------------------------------------------------------
+// Constant values for SemanticResourceAttributes
+//----------------------------------------------------------------------------------------------------------
+
+/**
+ * Name of the cloud provider.
+ */
+export const SEMRESATTRS_CLOUD_PROVIDER = (CLOUD_DOT +
+  'provider') as 'cloud.provider';
+
+/**
+ * The cloud account ID the resource is assigned to.
+ */
+export const SEMRESATTRS_CLOUD_ACCOUNT_ID = (CLOUD_DOT +
+  ACCOUNT_DOT +
+  ID) as 'cloud.account.id';
+
+/**
+ * The geographical region the resource is running. Refer to your provider&#39;s docs to see the available regions, for example [Alibaba Cloud regions](https://www.alibabacloud.com/help/doc-detail/40654.htm), [AWS regions](https://aws.amazon.com/about-aws/global-infrastructure/regions_az/), [Azure regions](https://azure.microsoft.com/en-us/global-infrastructure/geographies/), or [Google Cloud regions](https://cloud.google.com/about/locations).
+ */
+export const SEMRESATTRS_CLOUD_REGION = (CLOUD_DOT +
+  'region') as 'cloud.region';
+
+/**
+ * Cloud regions often have multiple, isolated locations known as zones to increase availability. Availability zone represents the zone where the resource is running.
+ *
+ * Note: Availability zones are called &#34;zones&#34; on Alibaba Cloud and Google Cloud.
+ */
+export const SEMRESATTRS_CLOUD_AVAILABILITY_ZONE = (CLOUD_DOT +
+  'availability_zone') as 'cloud.availability_zone';
+
+/**
+ * The cloud platform in use.
+ *
+ * Note: The prefix of the service SHOULD match the one specified in `cloud.provider`.
+ */
+export const SEMRESATTRS_CLOUD_PLATFORM = (CLOUD_DOT +
+  'platform') as 'cloud.platform';
+
+/**
+ * The Amazon Resource Name (ARN) of an [ECS container instance](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ECS_instances.html).
+ */
+export const SEMRESATTRS_AWS_ECS_CONTAINER_ARN = (AWS +
+  DOT +
```

Comment:
While the usage of the `DOT` and later `UNDERSCORE` variables seem like a waste because at the very least they cause all "usages" to expand from 1 character `.` to at least 2 (when minified) -- not counting the declaration of the variable in the first place.

This does allow "common" repeated strings to be reused more than once just because there is a leading / trailing `.` or `_` which taken in totality reduces the overall size of the static strings.

Reply:
This is also rationalized across both the SemanticAttributes.ts and SemanticResourceAttributes.ts, the "strings" where also chosen by hand and not automatically parsed to identify the longest common variants, which would have required some post-processing of the jinga template conversion or a LOT more work.